model/UniqueId/Service/UniqueIdSetter.php

```
@@ -0,0 +1,88 @@
+use oat\taoQtiItem\model\qti\Service;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class UniqueIdSetter
+{
+    private FeatureFlagCheckerInterface $featureFlagChecker;
+    private ResourceMetadataRetriever $resourceMetadataRetriever;
+    private ResourceUniqueIdRetriever $resourceUniqueIdRetriever;
+    private QtiIdentifierSetter $qtiIdentifierSetter;
+    private LoggerInterface $logger;
+
+    public function __construct(
+        FeatureFlagCheckerInterface $featureFlagChecker,
+        ResourceMetadataRetriever $resourceMetadataRetriever,
+        ResourceUniqueIdRetriever $resourceUniqueIdRetriever,
+        QtiIdentifierSetter $qtiIdentifierSetter,
+        LoggerInterface $logger
+    ) {
+        $this->featureFlagChecker = $featureFlagChecker;
+        $this->resourceMetadataRetriever = $resourceMetadataRetriever;
+        $this->resourceUniqueIdRetriever = $resourceUniqueIdRetriever;
+        $this->qtiIdentifierSetter = $qtiIdentifierSetter;
+        $this->logger = $logger;
+    }
+
+    public function __invoke(core_kernel_classes_Resource $item): core_kernel_classes_Resource
+    {
+        if (!$this->featureFlagChecker->isEnabled('FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER')) {
+            $this->logger->info('Unique identifier feature not enabled.');
+
+            return $item;
+        }
+
+        $originalResourceUri = $this->resourceMetadataRetriever->getOriginalResourceUri($item);
+
+        if ($originalResourceUri === null) {
+            $this->logger->info(sprintf('Cannot retrieve original resource URI for item %s', $item->getUri()));
+
+            return $item;
+        }
+
+        $originalResource = $item->getResource($originalResourceUri);
+        $uniqueIdentifier = $this->resourceUniqueIdRetriever->retrieve($originalResource);
+
+        if ($uniqueIdentifier === null) {
+            $this->logger->info(sprintf('Cannot retrieve unique identifier for item %s', $originalResource->getUri()));
```

- [ ] This should be an error exception. If this feature flag is active why would an Item/Test be missing this id and we masking the error?